test/transaction.js

```diff
@@ -224,66 +213,58 @@ function query(transaction) {
   };
 }
 
-function runTransaction(callback, request) {
-  let requests = Array.prototype.slice.call(arguments, 1);
-
-  firestore._firestoreClient._innerApiCalls.beginTransaction = function(
-      actual, options, callback) {
-    request = requests.shift();
-    assert.equal(request.type, 'begin');
-    assert.deepEqual(actual, request.request);
-    callback(request.error, request.response);
-  };
-
-  firestore._firestoreClient._innerApiCalls.commit = function(
-      actual, options, callback) {
-    request = requests.shift();
-    assert.equal(request.type, 'commit');
-    assert.deepEqual(actual, request.request);
-    callback(request.error, request.response);
-  };
-
-  firestore._firestoreClient._innerApiCalls.rollback = function(
-      actual, options, callback) {
-    request = requests.shift();
-    assert.equal(request.type, 'rollback');
-    assert.deepEqual(actual, request.request);
-    callback(request.error, request.response);
-  };
-
-  firestore._firestoreClient._innerApiCalls.batchGetDocuments = function(
-      actual) {
-    request = requests.shift();
-    assert.equal(request.type, 'getDocument');
-    assert.deepEqual(actual, request.request);
-    return request.stream;
-  };
-
-  firestore._firestoreClient._innerApiCalls.runQuery = function(actual) {
-    request = requests.shift();
-    assert.equal(request.type, 'query');
-    assert.deepEqual(actual, request.request);
-    return request.stream;
+function runTransaction(callback, ...requests) {
```

Comment:
What's the reason to support multiple requests now?

Reply:
These are the proto requests issued by the transaction that are used in the tests:

```
  it('support get with query', function() {
    return runTransaction((transaction, docRef) => {
      let query = docRef.parent.where('foo', '==', 'bar');
      return transaction.get(query).then(results => {
        assert.equal(results.docs[0].id, 'documentId');
      });
    }, begin(), query(), commit());
  });
```
In this case, we issue a beingTransaction, a query and a commit request.

I added a comment to explain this better.